src/repository/screens/repository.screen.js

```diff
@@ -207,14 +210,9 @@ class Repository extends Component {
     const { refreshing } = this.state;
 
     const initalRepository = navigation.state.params.repository;
-    const pulls = issues.filter(issue => issue.hasOwnProperty('pull_request')); // eslint-disable-line no-prototype-builtins
-    const pureIssues = issues.filter(issue => {
-      // eslint-disable-next-line no-prototype-builtins
-      return !issue.hasOwnProperty('pull_request');
-    });
 
-    const openPulls = pulls.filter(pull => pull.state === 'open');
-    const openIssues = pureIssues.filter(issue => issue.state === 'open');
+    const openPulls = pullRequests.filter(pull => pull.state === 'OPEN');
+    const openIssues = issues.filter(issue => issue.state === 'OPEN');
     const showFork =
```

Comment:
Using reactotron, I see that when we navigate to a repository screen, we pull 100 issues & 100 pull requests from the repo.
If the user doesn't click on "View All" for issues or PR, this is a waste of data & Github credit.

Could we have a single GraphQL query that retrieves only the 3 most recent & open issues & PR instead?

Reply:
Yes! I wondered about this, but was trying to translate directly from how the v3 API was being used to v4. I'll change the code that's slicing the first three results in JavaScript so that instead we just pass `first: 3` to the API.